### PR TITLE
Seed math/rand with UnixNano for truly random numbers.

### PR DESCRIPTION
**Pull request by paked**
While creating a random level generator for my game engine I noticed that each time I generated a level (this happens on startup) it was the same. This is a fix that removes this bug/issue. Perhaps this is desired behaviour, but I figure that this way otto will behave more like other javascript implementations, all of which have unpredictable random numbers.

Good Day!

### CODE REVIEW

File: otto.go [hunk 1]
```diff
@@ -225,9 +225,10 @@ package otto
 
 import (
 	"fmt"
-	"strings"
-
 	"github.com/robertkrimen/otto/registry"
+	"math/rand"
```
Comment (stevenh) [inline otto.go]: These entries should be above with fmt

**Comment by paked** [comment]
@stevenh Just fixed that up for you. Would you prefer the option to enable this put into a flag?

**Comment by stevenh** [comment]
Should always be done IMO.

My only question is should we create a rand object on the VM which is used, otherwise each VM is resetting the rand seed so might as well be done on init() instead of new VM.

**Comment by paked** [comment]
Ok, moved to ```init()```.

Also, browsing through the code. Perhaps the docs in ```otto.go``` should be moved into a ```doc.go```, I'd be happy to open another PR when this one gets closed with that change. (Or I could put it here, but that isn't particularly neat)

**Comment by dop251** [comment]
I wouldn't mess with the global rand, as it would interfere with the way other packages might want to seed it. Instead I would use a separate instance (rand.New(...)), seed it in the init() and potentially allow to override it using otto.SetRand().

**Comment by heavenlyhash** [comment]
There's a little better mechanism for a random seed:

```
var seed int64
binary.Read(realrand.Reader, binary.LittleEndian, &seed)
rnd = rand.New(rand.NewSource(seed))
```

This uses the best entropy available to the system, rather than the time (which is valid(ish) but should be a worst-case scenario fallback; most systems have better options).

**Comment by paked** [comment]
Ok, are we going with Eric's method? I'll implement that in the next few
hours,

**Comment by dop251** [comment]
It really depends on the application requirements. Sometimes random numbers don't need to be secure and in this case seeding with the current time works just fine. Using the secure source on the other hand could slow down the start when there is not enough entropy available.

Therefore I would use nano time by default and give the opportunity to override the generator.

**Comment by heavenlyhash** [comment]
This isn't secure regardless; that would require using `crypto/rand` through and through -- and that's not be particularly important since afaik the part of javascript we're replacing here never promised cryptographic rand.  I just feel bad seeding with nanotime; it's like it's 1999 and I'm writing php again where there wasn't any better option available.

As I read crypto/rand/rand_linux.go, it looks like the go authors Thought Of This and have done things that will consistently take the fastest possible path; it uses syscalls directly if available and nonblocking, and falls back to urandom reads otherwise, which are also nonblocking, even in the event of low entropy at boot (not that I would worry about that, unless you're replacing your init manager with a javascript runtime in golang for some reason).  In other words, no, it could *not* slow down the start.

tl;dr There is no downside to using the good seed.

**Comment by dop251** [comment]
You're right, according to the source code go doesn't use /dev/random pool in linux at all so it should not cause any significant slowdown.

However, going back to my initial point, I think it would be nice to have an ability to override the default PRNG.

**Comment by stevenh** [comment]
Making this configurable would indeed be preferable IMO

**Comment by deoxxa** [comment]
I fixed this via https://github.com/robertkrimen/otto/pull/163.
